### ISSUE TITLE
Preserve true order of images

### ISSUE THREAD

**Issue by stoyanovd** [sweeneychris/TheiaSfM#75]
I'm experimenting with building reconstruction in real time.
In this task I need a possibility to preserve true order of images (imitate receiving from stream).

Are you interesting in implementing this feature?
If not, I'll try to make it by myself a bit later, so I'm open for recommendations.

**Comment by sweeneychris** [comment]
By preserving the order, do you mean using incremental SfM but determining the order that images are added apriori?

**Comment by stoyanovd** [comment]
Yes, exactly this.

**Comment by sweeneychris** [comment]
What happens if you try to add the "next" image and it fails? For instance if the pose estimation cannot find enough inliers. Do you disregard it? Save it for later?

**Comment by stoyanovd** [comment]
At the moment - yes, assume we have only good situations, and only work with largest component.
Also with my current (simple) datasets, I have large overlay and can throw away one failed image.

**Comment by sweeneychris** [comment]
I see. I think there are enough concerns with instability in bundle adjustment and localization that I would rather not incorporate an interface for this into the incremental sfm solver. I'm happy to help you plan how you can implement it in your own branch of Theia though

**Comment by stoyanovd** [comment]
I agree with you. For sure it will be redundant for the entire theia...
So, I'll ask if run into problems =)

(sorry for close-reopen. I have misclicked while surfing github)
